ui/tests/integration/utils/client-count-utils-test.js

```diff
@@ -372,4 +373,30 @@ module('Integration | Util | client count utils', function (hooks) {
       'it formats combined data for monthly namespaces_by_key spanning upgrade to 1.10'
     );
   });
+
+  test('setStartTimeQuery: it returns start time query for activity log', async function (assert) {
+    assert.expect(5);
+    const apiPath = 'sys/internal/counters/config';
+    assert.strictEqual(setStartTimeQuery(true, {}), null, `it returns null if no permission to ${apiPath}`);
+    assert.strictEqual(
+      setStartTimeQuery(false, {}),
+      null,
+      `it returns null for community edition and no permission to ${apiPath}`
+    );
+    assert.strictEqual(
+      setStartTimeQuery(true, { billingStartTimestamp: new Date('2022-06-08T00:00:00Z') }),
+      1654646400,
+      'it returns unix time if enterprise and billing_start_timestamp exists'
+    );
+    assert.strictEqual(
+      setStartTimeQuery(false, { billingStartTimestamp: new Date('0001-01-01T00:00:00Z') }),
+      null,
+      'it returns null time for community edition even if billing_start_timestamp exists'
+    );
+    assert.strictEqual(
+      setStartTimeQuery(false, { foo: 'bar' }),
+      null,
+      'it returns null if billing_start_timestamp key does not exist'
```

Comment:
✨ these assertions are normally things i'd split into separate tests within a module. for example:
```typescript
module('#setStartTimeQuery', function (hooks) {
    test('community: it returns null with no permission to apiPath', async function (assert) {
      ...
    });

    test('community: it returns null when billing_start_time_exists', async function (assert) {
      ...
    });

    test('enterprise: it returns null when billing_start_time_exists', async function (assert) {
      ...
    }
  });
```

the main benefits of splitting them up this way are that it keeps the tests more isolated (and thus less flaky since there's no chance of the state of one assertion affecting the next one). it also makes them easier to read and debug (since you know exactly which situation failed).

not a blocker, since i know there's plenty of changes in this PR already but it's a pattern i'd encourage us to try out more 🙌

Reply:
That's a great idea! I'll keep this as is because it's how all of the tests in this file are formatted right now. I think I'd rather update all of them to be separate modules at once. 🤔

Great feedback, though. I'll keep this in mind when writing new tests and add a todo comment to the top of this file to keep it in mind as a future improvement.